presto-main/src/main/java/com/facebook/presto/operator/SetBuilderOperator.java

```diff
@@ -189,12 +193,30 @@ public boolean isFinished()
         return finished;
     }
 
+    /**
+     * This function processes unfinished work for the current page, and returns if it still
+     * needs input from the previous(child) operator.
+     * @return true if this operator needs additional input pages from its child, false otherwise
+     */
     @Override
     public boolean needsInput()
     {
+        // Since SetBuildOperator is always the last operator (no next operator that needs input from it),
+        // we handle the unfinished work in this method instead of getOutput(). The Driver always
+        // calls it in each iteration.
```

Comment:
Change to 
```
Since SetBuilderOperator doesn't produce any output, the getOutput()
method may never be called. We need to handle any unfinished work
before addInput() can be called again.
```

(Operator should not need to know about `Driver` and its behavioral patterns.)

Reply:
ok